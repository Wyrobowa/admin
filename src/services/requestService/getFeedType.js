/**
 * Returns object width url fragments based on passed argument
 * @param {string} feedType
 * @return {object}
 */
const getFeedType = (feedType) => {
  let feedTarget = {};

  switch (feedType) {
    case 'login':
      feedTarget = {
        target: '/login',
      };
      break;

    case 'mainMenu':
      feedTarget = {
        target: '/main-menu',
      };
      break;

    case 'checkAvailability':
      feedTarget = {
        target: '/check-availability',
      };
      break;

    case 'apartment':
      feedTarget = {
        target: '/admin/apartment',
      };
      break;

    case 'apartmentsList':
      feedTarget = {
        target: '/admin/apartments-list',
      };
      break;

    case 'location':
      feedTarget = {
        target: '/admin/location',
      };
      break;

    case 'locationsList':
      feedTarget = {
        target: '/admin/locations-list',
      };
      break;

    case 'page':
      feedTarget = {
        target: '/admin/page',
      };
      break;

    case 'pagesList':
      feedTarget = {
        target: '/admin/pages-list',
      };
      break;

    case 'language':
      feedTarget = {
        target: '/admin/language',
      };
      break;

    case 'languagesList':
      feedTarget = {
        target: '/admin/languages-list',
      };
      break;

    case 'phrase':
      feedTarget = {
        target: '/admin/phrase',
      };
      break;

    case 'phrasesList':
      feedTarget = {
        target: '/admin/phrases-list',
      };
      break;

    case 'translation':
      feedTarget = {
        target: '/admin/translation',
      };
      break;

    case 'translationsList':
      feedTarget = {
        target: '/admin/translations-list',
      };
      break;

    case 'apartmentFacility':
      feedTarget = {
        target: '/admin/apartment-facility',
      };
      break;

    case 'apartmentFacilityList':
      feedTarget = {
        target: '/admin/apartment-facility-list',
      };
      break;

    case 'apartmentFacilityGroup':
      feedTarget = {
        target: '/admin/apartment-facility-group',
      };
      break;

    case 'apartmentFacilityGroupList':
      feedTarget = {
        target: '/admin/apartment-facility-group-list',
      };
      break;

    case 'apartmentService':
      feedTarget = {
        target: '/admin/apartment-service',
      };
      break;

    case 'apartmentServiceList':
      feedTarget = {
        target: '/admin/apartment-service-list',
      };
      break;

    case 'apartmentServiceGroup':
      feedTarget = {
        target: '/admin/apartment-service-group',
      };
      break;

    case 'apartmentServiceGroupList':
      feedTarget = {
        target: '/admin/apartment-service-group-list',
      };
      break;

    case 'apartmentTranslations':
      feedTarget = {
        target: '/admin/apartment-translations',
      };
      break;

    case 'apartmentTranslationsList':
      feedTarget = {
        target: '/admin/apartment-translations-list',
      };
      break;

    case 'adminGalleryUpload':
      feedTarget = {
        target: '/admin/gallery/add',
      };
      break;

    case 'statics':
      feedTarget = {
        target: '/admin/statics',
      };
      break;

    default:
      feedTarget = {
        target: '',
      };
  }

  return feedTarget;
};

export default getFeedType;
