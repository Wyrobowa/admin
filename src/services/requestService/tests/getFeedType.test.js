import getFeedType from '../getFeedType';

describe('getFeedType', () => {
  it('should return valid object with params for login request', () => {
    expect(getFeedType('login')).toMatchSnapshot();
  });

  it('should return valid object with params for main-menu request', () => {
    expect(getFeedType('mainMenu')).toMatchSnapshot();
  });

  it('should return valid object with params for checkAvailability request', () => {
    expect(getFeedType('checkAvailability')).toMatchSnapshot();
  });
});
