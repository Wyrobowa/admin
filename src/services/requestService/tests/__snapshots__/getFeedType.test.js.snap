// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFeedType should return valid object with params for checkAvailability request 1`] = `
Object {
  "target": "/check-availability",
}
`;

exports[`getFeedType should return valid object with params for login request 1`] = `
Object {
  "target": "/login",
}
`;

exports[`getFeedType should return valid object with params for main-menu request 1`] = `
Object {
  "target": "/main-menu",
}
`;
